malariagen_data/ag3.py

```diff
@@ -314,6 +317,74 @@ def _prep_sample_sets_arg(self, *, sample_sets):
 
         return sample_sets
 
+    def _resolve_region(self, region):
+        """Parse the provided region and return `Region(contig, start, end)`.
+        Supports contig names, gene names and genomic coordinates"""
+
+        # region is already Region tuple
+        if isinstance(region, Ag3.Region):
+            return region
+
+        # search the geneset and match a genomic region regex pattern
+        gene_annotation = self.geneset(["ID"]).query(
+            f"type == 'gene' and ID == '{region}'"
+        )
+        region_pattern_match = re.search(r"([a-zA-Z0-9]+)\:(.+)\-(.+)", region)
+
+        # region is a chromosome arm
+        if region in self.contigs:
+            contig, start, end = region, None, None
+
+        # region is a gene name
+        elif not gene_annotation.empty:
+            gene_annotation = gene_annotation.squeeze()
+            contig = gene_annotation.contig
+            start = gene_annotation.start
+            end = gene_annotation.end
+
+        # parse region string that contains genomic coordinates
+        elif region_pattern_match:
+            region_split = region_pattern_match.groups()
+
+            contig = region_split[0]
+            start = int(region_split[1].replace(",", ""))
+            end = int(region_split[2].replace(",", ""))
+
+            if contig not in self.contigs:
+                raise ValueError(f"Contig {contig} does not exist in the dataset.")
+            elif (
+                start < 0
+                or end <= start
+                or end > self.genome_sequence(region=contig).shape[0]
+            ):
+                raise ValueError("Provided genomic coordinates are not valid.")
+
+        else:
+            raise ValueError(f"Region {region} is not valid.")
+
+        return Ag3.Region(contig, start, end)
+
+    def region_slice(self, region):
```

Comment:
```suggestion
    def locate_region(self, region):
```

Nit, minor naming suggestion.